src/components/List/ListItem.tsx

```diff
@@ -1,20 +1,30 @@
-import type { FC, PropsWithChildren } from 'react';
+import type { ComponentProps, FC } from 'react';
 import { twMerge } from 'tailwind-merge';
 import { mergeDeep } from '../../helpers/merge-deep';
 import { getTheme } from '../../theme-store';
 import type { DeepPartial } from '../../types';
 
 export interface FlowbiteListItemTheme {
-  base: string;
+  icon: string;
+  withIcon: {
+    on: string;
+    off: string;
+  };
 }
 
-export interface ListItemProps extends PropsWithChildren {
+export interface ListItemProps extends ComponentProps<'li'> {
   theme?: DeepPartial<FlowbiteListItemTheme>;
   className?: string;
+  Icon?: FC<ComponentProps<'svg'>>;
```

Comment:
usually props are lowercased, so for easier usage u can leave it lowercase and then destructure it like `icon: Icon` and voila!